pilot/pkg/model/sidecar.go

```diff
@@ -102,8 +103,10 @@ type SidecarScope struct {
 	// corresponds to a service in the services array above. When computing
 	// CDS, we simply have to find the matching service and return the
 	// destination rule.
-	destinationRules        map[host.Name][]*ConsolidatedDestRule
-	destinationRulesByNames map[types.NamespacedName]*config.Config
+	destinationRules       map[host.Name][]*ConsolidatedDestRule
+	destinationRulesByName map[types.NamespacedName]*config.Config
+
+	virtualServicesByName map[types.NamespacedName]*config.Config
```

Comment:
I think we may fall into a bad state, with every resource storred in sidecar, the memory cost is O(N*N)

I have no idea atm how to implement delta otherwise though

Reply:
Only VS and DR are the majority of them and will have in huge numbers. I would n't expect the same for other resources